core/src/main/java/com/nisovin/magicspells/listeners/CastListener.java

```diff
@@ -67,15 +69,29 @@ public void onPlayerInteract(PlayerInteractEvent event) {
 			return;
 		}
 
-		if (event.getAction() == Action.LEFT_CLICK_AIR || event.getAction() == Action.LEFT_CLICK_BLOCK) {
+		if ((event.getAction() == Action.LEFT_CLICK_AIR || event.getAction() == Action.LEFT_CLICK_BLOCK) && (!MagicSpells.areBowCycleButtonsReversed() || (event.hasItem() && event.getItem().getType() != Material.BOW && event.getItem().getType() != Material.CROSSBOW))) {
 			// Left click - cast
 			if (!MagicSpells.isCastingOnAnimate()) castSpell(event.getPlayer());
 			return;
 		}
 
-		if ((event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK) && (MagicSpells.isCyclingSpellsOnOffhandAction() || event.getHand() == EquipmentSlot.HAND)) {
+		//if ((event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK) && (MagicSpells.isCyclingSpellsOnOffhandAction() || event.getHand() == EquipmentSlot.HAND)) {
+		if ((event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK)
+				&& (!MagicSpells.areBowCycleButtonsReversed() || (event.hasItem() && event.getItem().getType() != Material.BOW && event.getItem().getType() != Material.CROSSBOW))
```

Comment:
Same here, 1.13 has to be supported for 4.0, also this if statement looks really scary now, you should work on it a bit more.